File: 2021/11/04/tribonacci.py
"""
https://leetcode.com/problems/n-th-tribonacci-number/
Author: Andrew Jarombek
Date: 11/4/2021
"""


def tribonacci(n: int) -> int:
    if n == 0:
        return 0

    if n < 3:
        return 1

    first = 0
    second = 1
    third = 1

    for i in range(3, n + 1):
        num = first + second + third
        first = second
        second = third
        third = num

    return third

File: 2022/10/02/tribonacci.ts
/**
 * https://leetcode.com/problems/n-th-tribonacci-number/
 * @author Andrew Jarombek
 * @since 10/2/2022
 */

function tribonacci(n: number): number {
    if (n == 0)
        return n;

    if (n <= 2)
        return 1;

    const seq = [0, 1, 1];
    let i = 3;

    while (i <= n) {
        let val = seq[0] + seq[1] + seq[2];
        seq[0] = seq[1];
        seq[1] = seq[2];
        seq[2] = val;

        i++;
    }

    return seq[2];
}